### PR TITLE
App page-business logic

### PR DESCRIPTION
**Pull request by lavanya-bmw**

### CODE REVIEW

File: cx-portal/src/components/pages/AppReleaseProcess/components/AppPage/index.tsx [hunk 1]
```diff
@@ -148,16 +178,16 @@ export default function AppPage() {
                       )}`,
                     },
                     pattern: {
-                      value: /^([A-Za-z.:@0-9&_ -]){10,200}$/,
+                      value: /^([A-Za-z.:@0-9&_ -]){10,2000}$/,
```
Comment (oyo) [inline cx-portal/src/components/pages/AppReleaseProcess/components/AppPage/index.tsx:181]: it was already there before but we have the `types/Patterns.ts` where we should manage all used Regex patterns in the application.

Comment (oyo) [inline cx-portal/src/components/pages/AppReleaseProcess/components/AppPage/index.tsx:181]: also there before but please use a const for the max length like `const MAX_INPUT_LENGTH = 2000` so we have only one place where to change this

File: cx-portal/src/components/pages/AppReleaseProcess/components/AppPage/index.tsx [hunk 2]
```diff
@@ -350,7 +380,7 @@ export default function AppPage() {
                   )} ${t('content.apprelease.appReleaseForm.isMandatory')}`,
                 },
                 pattern: {
-                  value: /^\+(\d{2})+(\(\s\d{3}\))?\s?\d{9,50}$/,
+                  value: Patterns.PHONE,
```
Comment (oyo) [inline cx-portal/src/components/pages/AppReleaseProcess/components/AppPage/index.tsx:383]: good

File: cx-portal/src/components/pages/AppReleaseProcess/components/ValidateAndPublish/index.tsx [hunk 3]
```diff
@@ -119,8 +120,8 @@ export default function VerifyCompanyData() {
     mode: 'onChange',
   })
 
-  const onCompanyDataSubmit = async (data: any) => {
-    setCompanyDataNotification(true)
+  const onValidatePublishSubmit = async (data: any) => {
+    setValidatePublishNotification(true)
   }
```
Comment (oyo) [inline cx-portal/src/components/pages/AppReleaseProcess/components/ValidateAndPublish/index.tsx:125]: are the curly brackets required here?

File: cx-portal/src/features/appManagement/slice.ts [hunk 4]
```diff
@@ -46,6 +50,8 @@ export const appManagementSelector = (state: RootState): SearchInputState =>
 export const currentActiveStep = (state: RootState): any =>
   state.management.currentActiveStep
 
+export const appIdSelector = (state: RootState): any => state.management.appId
```
Comment (oyo) [inline cx-portal/src/features/appManagement/slice.ts:53]: replace this `any` with the return type: `string`

**Review by oyo — CHANGES_REQUESTED** [review]
some minor remarks

**Comment by oyo** [comment]
please also update release notes

**Comment by lavanya-bmw** [comment]
> please also update release notes

this is already updated

**Review by oyo — APPROVED** [review]
looks good now

**Pull request merged**